File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:   
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:

        def isSame(node1, node2):
            if node1 is None and node2 is None:
                return True
            
            if node1 is None or node2 is None:
                return False

            return node1.val == node2.val and isSame(node1.left, node2.left) and isSame(node1.right, node2.right)
        
        queue = deque([root])

        while len(queue) != 0:
            cur = queue.popleft()

            if isSame(cur, subRoot):
                return True
            
            if cur.left:
                queue.append(cur.left)
            if cur.right:
                queue.append(cur.right)
        
        return False
            

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.java
class Solution {
    public int findMin(int[] nums) {
        int l = 0;
        int r = nums.length - 1;

        while (l < r) {
            int m = (r - l) + l / 2;

            if (nums[l] < nums[m]) { // left sorted
                l = m + 1;
            } else {
                r = m - 1;
            }
        }

        return nums[l];
    }
}
